src/daft-sketch/src/arrow2_serde.rs

```diff
@@ -0,0 +1,110 @@
+use arrow2::array::Array;
+use lazy_static::lazy_static;
+use serde_arrow::{
+    schema::{SchemaLike, SerdeArrowSchema, TracingOptions},
+    utils::{Item, Items},
+};
+use sketches_ddsketch::DDSketch;
+
+lazy_static! {
+    static ref ARROW2_DDSKETCH_ITEM_FIELDS: Vec<arrow2::datatypes::Field> =
+        SerdeArrowSchema::from_type::<Item<Option<DDSketch>>>(TracingOptions::default())
+            .unwrap()
+            .to_arrow2_fields()
+            .unwrap();
+
+    /// The corresponding arrow2 DataType of Vec<DDSketch> when serialized as an arrow2 array
+    pub static ref ARROW2_DDSKETCH_DTYPE: arrow2::datatypes::DataType = ARROW2_DDSKETCH_ITEM_FIELDS.first().unwrap().data_type().clone();
+}
+
+/// Converts a Vec<Option<DDSketch>> into an arrow2 Array
+pub fn into_arrow2(sketches: Vec<Option<DDSketch>>) -> Box<dyn arrow2::array::Array> {
```

Comment:
you should probably take in a reference or slice here so you can avoid a redundant clone